src/main/java/com/eventsphere/user/controller/UserController.java

```diff
@@ -0,0 +1,151 @@
+package com.eventsphere.user.controller;
+
+import com.eventsphere.user.model.User;
+import com.eventsphere.user.model.dto.ChangePasswordDto;
+import com.eventsphere.user.model.dto.UserDto;
+import com.eventsphere.user.service.UserService;
+import jakarta.validation.Valid;
+import lombok.RequiredArgsConstructor;
+import org.springframework.hateoas.CollectionModel;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.servlet.support.ServletUriComponentsBuilder;
+
+import java.net.URI;
+import java.util.List;
+
+import static org.springframework.hateoas.server.mvc.WebMvcLinkBuilder.linkTo;
+import static org.springframework.hateoas.server.mvc.WebMvcLinkBuilder.methodOn;
+
+/**
+ * Controller class for managing user-related operations.
+ */
+@RestController
+@RequestMapping("v1/users")
+@RequiredArgsConstructor
+public class UserController {
+
+    private static final String GET_USER_REL = "get-user";
+    private static final String CHANGE_PASSWORD_REL = "change-password";
+    private static final String CREATE_USER_REL = "create-user";
+    private static final String GET_ALL_USERS_REL = "get-all-users";
+    private static final String SELF_REL = "self";
+    private final UserService userService;
+
+    /**
+     * Retrieves a list of all users.
+     *
+     * @return ResponseEntity with the list of users and HTTP status OK.
+     */
+    @GetMapping
+    public ResponseEntity<CollectionModel<User>> getAllUsers() {
+        List<User> users = userService.getAll();
+
+        for (User user : users) {
+            user.add(
+                    linkTo(methodOn(UserController.class).getUser(user.getId())).withRel(GET_USER_REL),
```

Comment:
what is happening here ?

Reply:
Adding HATEOAS links to every user, so it will be easy to get the exact one.